File: CJ19/1A/a.py
t = int(input())
r = 0
c = 0
vis = []

def dfs(x, y, tot):
	if tot == r*c:
		print ("POSSIBLE")
		print (x+1, y+1)
		return True
	vis[x][y] = 1
	for i in range(r):
		for j in range(c):
			if not vis[i][j]:
				a = i != x
				a &= j != y
				a &= (i-j) != (x-y)
				a &= (i+j) != (x+y)
				if a:
					if dfs(i, j, tot+1):
						print(x+1, y+1)
						return True
	vis[x][y] = 0
	return False

for x in range(t):
	print("Case #{}: ".format(x+1), end='')

	r, c = map(int, input().split())

	vis = []
	for x in range(r):
		vis.append([])
		for y in range(c):
			vis[x].append(0)



	if not dfs(0, 0, 1):
		print ("IMPOSSIBLE")

File: CJ19/1A/a.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    int t;
    cin >> t;
    for(int caso = 1; caso <= t; ++caso){
        int r, c; cin >> r >> c;
        bool vuelta = r > c;
        if (vuelta)
            swap(r, c);
        cout << "Case #" << caso << ": ";
        int d;
        for (d = 2; d < c - 1; d++){
            int tot = r*c;
            int x = 0, y = 0;
            vector<vector<int>> mat(r, vector<int>(c, 0));
            while (tot){
                if (mat[x][y] == 1){
                    break;
                }
                mat[x][y] = 1;
                tot--;
                x += 1;
                x %= r;
                
                y += d;
                y %= c;
                
            }
            if (!tot){
                goto solve;
            }
        }
        cout << "IMPOSSIBLE" << endl;
        continue;
        
        solve:
        cout << "POSSIBLE" << endl;
        int tot = r*c;
        int x = 0, y = 0;
        
        while (tot){
            if (vuelta){
                cout << y+1 << " " << x + 1 << endl;
            }else{
                cout << x+1 << " " << y + 1 << endl;
            }
            tot--;
            x += 1;
            x %= r;
            
            y += d;
            y %= c;
            
        }
    }
    
}

/*

|x| | | |
| | |x| |
|x| | | |


| | | | |

*/